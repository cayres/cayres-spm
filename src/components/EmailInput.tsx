import React from "react";
import { StyleProp, Text, TextInput, TextStyle, View } from "react-native";

export interface EmailProps {
    textInputStyle: StyleProp<TextStyle>;
    value: string;
    onChangeText: (text: string) => void;
}

interface EmailState {
    valid: boolean;
}

export const isValidEmail = (email: string) => {
    return email.match(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
};

export class EmailInput extends React.PureComponent<EmailProps, EmailState> {

    constructor(props: EmailProps) {
        super(props);
        this.state = {
            valid: true,
        };

        this.onHandleBlur = this.onHandleBlur.bind(this);
        this.onHandleFocus = this.onHandleFocus.bind(this);
    }

    public render(): React.ReactNode {
        const {textInputStyle, value, onChangeText} = this.props;
        const { valid } = this.state;

        const erro = valid ? false : <Text>E-mail inv√°lido</Text>;

        return (
            <View>
                <TextInput
                    placeholder="E-mail"
                    value={value}
                    onChangeText={onChangeText}
                    style={valid ? textInputStyle : [textInputStyle, {color: "red"}]}
                    onBlur={this.onHandleBlur}
                    onFocus={this.onHandleFocus}
                    autoCapitalize="none"
                />
                {erro}
            </View>
        );
    }

    private onHandleBlur() {
        const { value } = this.props;

        if (value.length > 0 && !isValidEmail(value)) {
            this.setState({valid: false});
        }
    }

    private onHandleFocus() {
        this.setState({valid: true});
    }
}
