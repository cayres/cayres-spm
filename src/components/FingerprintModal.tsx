import React from "react";
import { Image, Modal, StyleSheet, Text, View } from "react-native";
import { Dialog, DialogDefaultActions } from "react-native-material-ui";

interface Props {
    visible: boolean;
    onRequestClose: any;
}

const FingerprintPopup: React.SFC<Props> = (props) => {

    const {visible, onRequestClose} = props;

    return (
        <Modal animationType="slide" visible={visible} transparent={true} onRequestClose={onRequestClose}>
            <View style={styles.container}>
                <Dialog>
                    <Dialog.Title><Text>Authenticate with Touch</Text></Dialog.Title>
                    <Dialog.Content>
                    <   Image style={styles.image} source={require("../assets/finger_print.png")} />
                    </Dialog.Content>
                    <Dialog.Actions>
                    <DialogDefaultActions
                        actions={["cancel"]}
                        onActionPress={onRequestClose}
                        style={{defaultActionsContainer: {backgroundColor: "#FFFFFF"}}}
                    />
                    </Dialog.Actions>
                </Dialog>
            </View>
        </Modal>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
    },
    image: {
        alignSelf: "center",
    },
});

export default FingerprintPopup;
