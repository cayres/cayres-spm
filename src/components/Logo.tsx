import React from "react";
import { Image, StyleSheet, View } from "react-native";

export const Logo: React.SFC<{}> = (props) => {
    return (
        <View>
            <Image source={require("../img/logo-cedro.png")}/>
        </View>
    );
};
