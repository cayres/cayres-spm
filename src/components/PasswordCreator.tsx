import React, { SFC } from "react";
import { Image, StyleProp, Text, TextInput, TextStyle, View } from "react-native";

export interface IPasswordProps {
    textInputStyle: StyleProp<TextStyle>;
    value: string;
    onChangeText: (text: string) => void;
}

export const isValidPassword = (password: string) => {
    return password.match(/(?=(.*[0-9]))(?=.*[\!@#$%^&*()\\[\]{}\-_+=~`|:;"'<>,./?])(?=(.*[\w]))(?=(.*)).{8,}/g) && password.indexOf(" ") === -1;
};

export const PasswordCreator: React.SFC<IPasswordProps> = (props) => {
    const {textInputStyle, value, onChangeText} = props;

    const alert = isValidPassword(value)  ? false : <Image source={require("../img/report_problem.png")} />;

    return (
        <View style={{alignItems: "flex-end"}}>
            <TextInput
                placeholder="Senha"
                secureTextEntry={true}
                value={value}
                onChangeText={onChangeText}
                style={textInputStyle}
            />
            <View style={{flexDirection: "row" }}>
                <Text style={{width: 250, fontSize: 12 }}>{"Use oito ou mais caracteres com uma combinação de letras, números e símbolos "}</Text>
                <Text>{`8/${value.length} `}</Text>
                {alert}
            </View>
        </View>
    );
};
