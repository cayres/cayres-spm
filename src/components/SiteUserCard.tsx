import React from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import { Card } from "react-native-material-ui";

interface SUCardProps {
    token: string;
    url: string;
    user: string;
}

const SiteUserCard: React.SFC<SUCardProps> = (props) => {

    const {token, url, user} = props;

    const site = url.split("://")[1];

    return (
        <Card>
            <View style={styles.box}>
                <Image
                    source={{
                        uri: `https://dev.people.com.ai/mobile/api/v2/logo/${site}`,
                        method: "GET",
                        headers: {
                            Authorization: token,
                        },
                    }}
                    style={styles.image}
                />
                <View style={styles.textView}>
                    <Text style={styles.siteText}>{site.toLocaleLowerCase()}</Text>
                    <Text style={styles.userText}>{user}</Text>
                </View>
            </View>
        </Card>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    box: {
        padding: 5,
        flexDirection: "row",
        alignContent: "center",
        alignItems: "center",
    },
    image: {
        width: 50,
        height: 50,
        resizeMode: "contain",
    },
    textView: {paddingLeft: 20, alignSelf: "center"},
    siteText: {color: "#2F4E78", fontSize: 12},
    userText: {color: "#2F4E78", fontSize: 18},
});

export default SiteUserCard;
