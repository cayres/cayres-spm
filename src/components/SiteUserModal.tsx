import React from "react";
import { Image, Modal, StyleSheet, TextInput, View } from "react-native";
import { Button } from "react-native-material-ui";

interface SUModalProps {
    token: string;
    data?: SUModalState;
    handleSave: (value: SUModalState) => void;
    handleDelete: (id: number) => void;
}

export interface SUModalState {
    url?: string;
    user?: string;
    password?: string;
    id?: number;
}

interface ModalState {
    security: boolean;
    visible: boolean;
}

const INITIAL_STATE: SUModalState = {
    url: "",
    password: "",
    user: "",
    id: 0,
};

class SiteUserModal extends React.PureComponent<SUModalProps, SUModalState & ModalState> {

    constructor(props: SUModalProps) {
        super(props);
        this.state = {
            ...INITIAL_STATE,
            security: true,
            visible: false,
        };

        this.changeSecurity = this.changeSecurity.bind(this);
        this.open = this.open.bind(this);
        this.close = this.close.bind(this);
        this.onClickSave = this.onClickSave.bind(this);
        this.onClickDelete = this.onClickDelete.bind(this);
    }

    public render() {
        const { token } = this.props;
        const { password, url, user, security, visible } = this.state;

        const site = url!.indexOf("://") > -1 ? url!.split("://")[1] : url;

        return (
            <Modal animationType="slide" visible={visible} onRequestClose={() => this.close()}>
                <View style={styles.container}>
                    <Image source={{
                        uri: `https://dev.people.com.ai/mobile/api/v2/logo/${site ? site : "unknow"}`,
                        method: "GET",
                        headers: {
                            Authorization: token,
                        },
                    }} style={styles.image} />
                    <View style={styles.form}>
                        <TextInput
                            style={styles.input}
                            placeholder={"Site"}
                            value={url}
                            key="site"
                            autoCapitalize="none"
                            onChangeText={(text) => this.setState({url: text.trim()})}
                        />
                        <TextInput
                            style={styles.input}
                            placeholder={"User"}
                            value={user}
                            onChangeText={(text) => this.setState({user: text})}
                        />
                        <View style={styles.buttonView}>
                            <TextInput
                                style={[styles.input, { width: 240 }]}
                                placeholder={"Password"}
                                secureTextEntry={security}
                                value={password}
                                onChangeText={(text) => this.setState({password: text})}
                            />
                            <Button onPress={this.changeSecurity} text="" icon={security ? "visibility" : "visibility-off"} />
                        </View>
                    </View>
                    <View style={styles.buttonView}>
                        <Button onPress={() => this.close()} text="" icon="arrow-back" />
                        <Button accent onPress={this.onClickDelete} text="" icon="delete" />
                        <Button onPress={this.onClickSave} text="" icon="done" style={{text: {color: "#8FC74B"}}} />
                    </View>
                </View>
            </Modal>
        );
    }

    public open(value?: SUModalState) {
        this.setState({visible: true, ...value});
    }

    public close() {
        this.setState({visible: false, ...INITIAL_STATE});
    }

    private changeSecurity() {
        const { security } = this.state;
        this.setState({security: !security});
    }

    private onClickSave(): void {
        const { security, visible, ...value} = this.state;
        const { handleSave } = this.props;
        this.close();
        handleSave(value);
    }

    private onClickDelete(): void {
        const { id } = this.state;
        const { handleDelete } = this.props;
        this.close();
        handleDelete(id!);
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: "column",
        justifyContent: "space-around",
        alignItems: "center",
    },
    form: {
        alignItems: "center",
        alignSelf: "center",
    },
    input: {
        width: 300,
        maxHeight: 50,
        color: "#2F4E78",
        marginBottom: 10,
    },
    image: {
        width: 100,
        height: 100,
        resizeMode: "contain",
    },
    buttonView: {
        flexDirection: "row",
        alignItems: "baseline",
    },
});

export default SiteUserModal;
