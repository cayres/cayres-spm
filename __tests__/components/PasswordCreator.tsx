import React from 'react';
import renderer from 'react-test-renderer';

import {PasswordCreator} from '../../src/components/PasswordCreator';

it('renders correctly with defaults', () => {
  const input = renderer
    .create(<PasswordCreator value="123" onChangeText={(text => console.log(text))} textInputStyle={{}}  />)
    .toJSON();
  expect(input).toMatchSnapshot();
});