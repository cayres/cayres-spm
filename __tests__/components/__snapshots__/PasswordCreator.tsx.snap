// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      0="1"
      1="2"
      2="3"
      allowFontScaling={true}
      maxLength={8}
      placeholder="Senha"
      secureTextEntry={true}
      style={Object {}}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    3/8
  </Text>
</View>
`;
